src/config.c

```diff
@@ -299,7 +299,8 @@ bool config_to_color(const char* text, argb_t* color)
         ++text;
     }
 
-    if (!str_to_num(text, 0, &num, 16) || num < 0 || num > 0xffffffff) {
+    if (!str_to_num(text, 0, &num, 16) || num < 0 ||
+        (uint64_t)num > (uint64_t)0xffffffff) {
```

Comment:
Do we really need these type casts?
`0xffffffff` fits uint32_t. Also color is always 32 bits.

Reply:
On 32-bit systems, `0xffffffff` will be considered a 4-byte `unsigned int` by compilers, as it can't fit in an `int` without overflowing. Also, `ssize_t num` will be a 4-byte `int` (not guaranteed by the C spec, but what compilers usually do). The compiler will emit a warning because these two different types are being compared, and the compiler doesn't know whether it should generate a signed or unsigned comparison. The cast to `uint64_t` assures that these two values can be safely compared on both 32-bit and 64-bit systems. On 64-bit systems, because this code checks for the validity of a color, `num` can be greater than 32-bits (it's inputted by the user in the configuration file). A cast of `num` to `uint32_t` wouldn't suffice, as values higher than `0xffffffff` would be accepted as valid colors, as they'd be truncated before the comparison.